app/src/main/java/de/westnordost/streetcomplete/quests/address/AddAddressStreet.kt

```diff
@@ -0,0 +1,108 @@
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.OsmTaggings
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataAndGeometryDao
+import de.westnordost.streetcomplete.data.osm.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.ElementPolylinesGeometry
+import de.westnordost.streetcomplete.data.osm.tql.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.osm.tql.toGlobalOverpassBBox
+import de.westnordost.streetcomplete.quests.localized_name.data.RoadNameSuggestionsDao
+import java.util.regex.Pattern
+
+class AddAddressStreet(
+        private val overpassServer: OverpassMapDataAndGeometryDao,
+        private val roadNameSuggestionsDao: RoadNameSuggestionsDao
+) : OsmElementQuestType<AddressStreetAnswer> {
+    override val commitMessage = "Add street/place names to address"
+    override val icon = R.drawable.ic_quest_housenumber_street
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_address_street_title
+
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        val housenumber = tags["addr:housenumber"]
+        return if (housenumber != null) arrayOf(housenumber) else arrayOf()
+    }
+
+    override fun createForm() = AddAddressStreetForm()
+
+    override fun isApplicableTo(element: Element): Boolean? = null
+
+    override fun download(bbox: BoundingBox, handler: (element: Element, geometry: ElementGeometry?) -> Unit): Boolean {
+        Log.wtf("aaa", getOverpassQuery(bbox))
+        if (!overpassServer.query(getOverpassQuery(bbox), handler)) return false
+        if (!overpassServer.query(getStreetNameSuggestionsOverpassQuery(bbox), this::putRoadNameSuggestion)) return false
+        return true
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox) =
+            bbox.toGlobalOverpassBBox() + """
+            relation["type"="associatedStreet"];
+            > -> .inStreetRelation;
+
+            way["highway" ~ "^(${OsmTaggings.ALL_ROADS.joinToString("|")})$"]["name"] -> .named_roads;
+
+            nwr["addr:street"!~".*"]["addr:housenumber"]["addr:place"!~".*"](around.named_roads:150) -> .missing_data;
+
+            (.missing_data; - .inStreetRelation;);""" +
+                    getQuestPrintStatement()
+
+    // TODO: probably either this should be modified to match AddRoadName with custom set of roads  with config array migrated to OsmTaggings
+    // TODO: or AddRoadName can use something like that
+    /** return overpass query string to get roads with names
+     * */
+    private fun getStreetNameSuggestionsOverpassQuery(bbox: BoundingBox) =
+            bbox.toGlobalOverpassBBox() + "\n" + """
+        way[highway ~ "^(${OsmTaggings.ALL_ROADS.joinToString("|")})$"][name];
+        out body geom;""".trimIndent()
```

Comment:
`AddRoadName` only needs the names of roads that are near unnamed roads.
`AddAddressStreet` needs the names of roads that are near housenumbers with missing `address:street`.

The suggestions are put into the same table, but it makes sense if the queries query a different set of roads to only put into the road suggestions db what is really necessary. (Because currently that db table is *never* cleared or cleaned up.)

Anyway, actions:
1. Compare `AddRoadName.ROADS` and `OsmTaggings.ALL_ROADS` and use `OsmTaggings.ALL_ROADS` also in `AddRoadName` in place of `ROADS` if it is the same
2. kind of optional, but it would be better: analogous to `AddRoadName`, `AddAddressStreet` should only query the names of roads that are near housenumber nodes that are missing a road name.

Reply:
Ok, I've made changes to consolidate the way that name suggestions work here and in AddRoadName to meet (1). I've also changes the query to only look at roads around addresses without streets to meet (2). I've added some comments below to point out where I've done this.